### DISCUSSION TITLE
Build Hyprland without systemd using nix build on voidlinux

### DISCUSSION THREAD

**Discussion by moilong** [hyprwm/Hyprland#4315] [General]
I built Hyprland by editing [Hyprland/blob/main/nix/default.nix](url) as
`{withSystemd ? false,}`

`mesonFlags = builtins.concatLists [
(lib.optional withSystemd "-Dsystemd=disabled")
  ];`

Then run `nix build /path/to/Hyprland local repo`
Built version:
`void@mnbvcxz ~> hyprctl version
Hyprland, built from branch  at commit  dirty ().
Date: 2023-12-30
Tag: 
flags: (if any)`
Is the steps I used is correct to build Hyprland without systemd using nix? How can I make sure it is built with no systemd?

**Comment by fufexan** [comment]
> How can I make sure it is built with no systemd?

Try `nix why-depends <hyprland-path> nixpkgs#systemd^out`

**Reply by moilong** [reply]
void@mnbvcxz ~> nix why-depends /nix/store/y0yg38js9jxrwwxsw5h3n00dhspj55ss-hyprland-0.34.0+date=2024-01-02_1512b81/bin/.Hyprland-wrapped nixpkgs#systemd^out
/nix/store/y0yg38js9jxrwwxsw5h3n00dhspj55ss-hyprland-0.34.0+date=2024-01-02_1512b81
└───/nix/store/wzzr1zzx34qvr94zr87xh0nr48zmfg4n-wlroots-hyprland-2023-12-30_f81c3d9
    └───/nix/store/ksghzkn5gxi0vm48m39jqd1xiv2h4yp4-seatd-0.8.0
        └───/nix/store/iidxwcyp8pqhrq3iji17shs4m6gin0kv-systemd-254.6

So this means Hyprland is built with systemd?

**Reply by fufexan** [reply]
Yes, looks like it.
